frontend/src/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useRef, useState } from 'react';
+
+interface UseIntersectionObserverProps {
+  threshold?: number;
+}
+
+const useIntersectionObserver = ({
+  threshold = 0.5,
+}: UseIntersectionObserverProps) => {
+  const observer = useRef<IntersectionObserver | null>(null);
+  const targetRef = useRef<HTMLDivElement | null>(null);
+  const [isIntersecting, setIsIntersecting] = useState(false);
+
+  useEffect(() => {
+    if (!targetRef.current) return;
```

Comment:
브라우저 호환성 체크를 위해 이런 코드를 추가해보면 어떨까요?
검색해보니 `IntersectionObserver`를 지원하지 않는 브라우저가 있네요. 
혹시 모를 에러를 방지하기 위해 있으면 좋을 것 같다는 생각이 드네요~!

- Internet Explorer: 모든 버전에서 지원 안 함
- Safari: 12.1 이전 버전 (2019년 3월 이전)
- Chrome: 51 이전 버전 (2016년 5월 이전)
- Firefox: 55 이전 버전 (2017년 8월 이전)

```suggestion
  useEffect(() => {
  
    if (!('IntersectionObserver' in window)) {
        console.warn('IntersectionObserver is not supported in this browser');
        return;
      }
  
    if (!targetRef.current) return;
```

Reply:
찾아보니 intersection observer에 대응 가능한 폴리필이 존재하네요! (구글에서 제작)
[intersection-observer](https://www.npmjs.com/package/intersection-observer)

우선은 루나가 알려준 대로 경고 처리를 해둔 후,
추후 유지 보수에서 기술 검토 후에 도입해보면 좋을 것 같아요!
꼼꼼한 리뷰 감사합니다 :)